protocols/ping/src/lib.rs

```diff
@@ -167,4 +159,33 @@ impl NetworkBehaviour for Behaviour {
             | FromSwarm::ExpiredExternalAddr(_) => {}
         }
     }
+
+    fn on_connection_handler_event(
+        &mut self,
+        peer: PeerId,
+        _: ConnectionId,
+        result: THandlerOutEvent<Self>,
+    ) {
+        self.events.push_front(Event { peer, result })
+    }
```

Comment:
Note to self: Check that this diff gets better once we merge https://github.com/libp2p/rust-libp2p/pull/3368.